src/points/PointsHistoryBottomSheet.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useMemo } from 'react'
+import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'
+import SectionHead from 'src/components/SectionHead'
+import GorhomBottomSheet from '@gorhom/bottom-sheet'
+import { useDispatch, useSelector } from 'src/redux/hooks'
+import { pointsHistoryStatusSelector, pointsHistorySelector } from 'src/points/selectors'
+import { useTranslation } from 'react-i18next'
+import { typeScale } from 'src/styles/fonts'
+import BottomSheetBase from 'src/components/BottomSheetBase'
+import { Spacing } from 'src/styles/styles'
+import Attention from 'src/icons/Attention'
+import Colors from 'src/styles/colors'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { PointsEvents } from 'src/analytics/Events'
+import { getHistoryStarted } from 'src/points/slice'
+import { ClaimHistory } from 'src/points/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import colors from 'src/styles/colors'
+import { BottomSheetSectionList } from '@gorhom/bottom-sheet'
+
+interface Props {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+}
+
+function PointsHistoryBottomSheet({ forwardedRef }: Props) {
+  const { t } = useTranslation()
+
+  const dispatch = useDispatch()
+
+  const pointsHistoryStatus = useSelector(pointsHistoryStatusSelector)
+  const pointsHistory = useSelector(pointsHistorySelector)
+
+  const onPressTryAgain = () => {
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_try_again_press)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: false,
+      })
+    )
+  }
+
+  const onFetchMoreHistory = () => {
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_fetch_more)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: true,
+      })
+    )
+  }
+
+  const sections = useMemo(() => {
+    if (!pointsHistory.length || pointsHistoryStatus === 'error') {
+      return []
+    }
+    return groupFeedItemsInSections([], pointsHistory, (t: ClaimHistory) => Date.parse(t.createdAt))
+  }, [pointsHistory, pointsHistoryStatus])
+
+  // TODO: Render items
+  const renderItem = () => {
+    return <></>
+  }
+
+  const renderLoading = () => {
```

Comment:
We should avoid defining components inline. it's bad for perf and can have unintended effects (like if we use `useEffects` in them they'll be triggered on each render, since the parent comp is created on each reader).

Reply:
Got it, I've pulled these out into their own functional components.